scripts/lib/CIME/case/check_input_data.py

```diff
@@ -6,9 +6,45 @@
 from CIME.XML.inputdata import Inputdata
 import CIME.Servers
 
-import glob
+import glob, hashlib
 
 logger = logging.getLogger(__name__)
+# The inputdata_checksum.dat file will be read into this hash if it's available
+chksum_hash = dict()
+
+
+def _download_checksum_file(server, input_data_root):
+    """
+    Return True if successfully downloaded
+    """
+    rel_path = "../inputdata_checksum.dat"
```

Comment:
Is it a problem that there is only one checksum file despite potentially having multiple inputdata servers? I would imagine having a separate checksum file for each server, with different file names distinguishing them. But maybe I'm missing something here.

Maybe this isn't a huge deal, because (if I understand the operation of this code correctly), a given call to `check_input_data` only works with a single remote server, and so would create a new `inputdata_checksum.dat` for that server. But it seems like this could still create some confusion, especially if there are any problems.

I don't feel strongly that this needs to be fixed, but wanted to run this by you.

Reply:
I agree that this could be a problem, I've dealt with it as follows:  The server checksum file is now listed in config_inputdata.xml and the client checksum file is a unique merge of server checksum files.   If the same file has different checksums on different servers an error will result.

Reply:
Your solution seems good - thanks.